tests/unit/rankers/test_algorithms.py

```diff
@@ -295,10 +301,10 @@ def test_rank_by_ctr_over_n_trials(self, thompson_sampling_function, metrics, nt
         assert final_ranks[3][0] == '222222'
 
         # ranks are not deterministic
-        assert int(ranks['999999']) != ranks['999999']
-        assert int(ranks['666666']) != ranks['666666']
-        assert int(ranks['333333']) != ranks['333333']
-        assert int(ranks['222222']) != ranks['222222']
+        assert min_ranks['999999'] != max_ranks['999999']
+        assert min_ranks['666666'] != max_ranks['666666']
+        assert min_ranks['333333'] != max_ranks['333333']
+        assert min_ranks['222222'] != max_ranks['222222']
```

Comment:
Asserting that the minimum rank is different from the maximum I think is closer to the intention. Even this failed occasionally, but by increasing ntrails the test became stable.